pkg/tekton/syntax/pipeline.go

```diff
@@ -1211,7 +1212,7 @@ func (s *Step) modifyStep(params StepPlaceholderReplacementArgs) {
 			sourceDir := params.WorkspaceDir
 			dockerfile := filepath.Join(sourceDir, "Dockerfile")
 			localRepo := params.DockerRegistry
-			destination := params.DockerRegistry + "/" + params.DockerRegistryOrg + "/" + params.GitName
+			destination := params.DockerRegistry + "/" + params.DockerRegistryOrg + "/" + naming.ToValidName(params.GitName)
```

Comment:
Is there a way to add a test for this in some form? Either as unit test or even as part of the bdd tests for eks (no idea where this stands atm).

@abayer is the expert on testing this syntax stuff and @dgozalo might now how we stand on bdd tests.

Reply:
Probably the best way to do this would be in https://github.com/jenkins-x/jx/blob/master/pkg/cmd/step/create/step_create_task_test.go - specifically to change https://github.com/jenkins-x/jx/blob/master/pkg/cmd/step/create/step_create_task_test.go#L86-L93 to have a `.` in its `repoName`, like `js.test-repo`. That'll fail initially due to the content in https://github.com/jenkins-x/jx/tree/master/pkg/cmd/step/create/test_data/step_create_task/js_build_pack not matching, but you can then update the various files in there to use `js.test-repo` rather than `js-test-repo` for the repo (just search for `js-test-repo` in that directory and change everything that isn't the `--destination=...` flag for a kaniko step in `tasks.yml`).

Yes, this ideally would be tested from `pipeline_test.go`, but this is messier than it should be.

Reply:
And fwiw, I do think it's fine to just modify the existing test case - there are so many test cases that end up overlapping in the same areas that tweaking an existing test here isn't going to decrease coverage. The "no `.`s in the repo name, replacing skaffold in the build pack with kaniko in the Tekton CRDs" case is already covered in basically the exact same way in the `maven_build_pack` test case, for example.